registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/test/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/ExceptionBiometricsTagGeneratorTest.java

```diff
@@ -1,123 +1,205 @@
+		throws BaseCheckedException {
+		Map<String, String> metaInfoMap = new HashMap<>();
+		Map<String, String> tags = 
+			exceptionBiometricsTagGenerator.generateTags("12345", "1234", "NEW", null, metaInfoMap, 0);
+		assertEquals(notAvailableTagValue, tags.get(tagName));
+	}
+	
+	@Test
+	public void getRequiredIdObjectFieldNamesTest() throws Exception {
+		List<String> result = exceptionBiometricsTagGenerator.getRequiredIdObjectFieldNames();
+		assertEquals(result, null);
+	}
+
+	@Test
+	public void testGenerateTagsForExceptionBiometricsAvailableWithBCTest()
+			throws BaseCheckedException, JsonMappingException, JsonProcessingException {
+		Map<String, String> metaInfoMap = new HashMap<>();
+		metaInfoMap.put(JsonConstant.EXCEPTIONBIOMETRICS,
+				"{\"applicant\" : {\"leftEye\" : {\"type\" : \"iris\", \"missingBiometric\" : \"leftEye\",\"reason\" : \"Missing biometrics\",\"exceptionType\" : \"Permanent\",\"individualType\" : \"INDIVIDUAL\"}}}");
+		metaInfoMap.put(JsonConstant.METADATA,
+				"[ {\n  \"label\" : \"registrationId\",\n  \"value\" : \"1234\"\n},{\n  \"label\" : \"Registration Client Version Number\",\n  \"value\" : \"1.1.4\"\n} ]");
+		Mockito.when(mapper.readValue(anyString(), any(Class.class)))
+				.thenReturn(new FieldValue("Registration Client Version Number", "1.1.4"));
+		Map<String, String> tags = exceptionBiometricsTagGenerator.generateTags("12345", "1234", "NEW", null,
+				metaInfoMap, 0);
+		assertEquals("LE", tags.get(tagName));
+	}
+
+	@Test
+	public void testGenerateTagsForExceptionBiometricsAvailableWithoutversion()
+			throws BaseCheckedException, JsonMappingException, JsonProcessingException {
+		Map<String, String> metaInfoMap = new HashMap<>();
+		metaInfoMap.put(JsonConstant.EXCEPTIONBIOMETRICS,
+				"{\"individualBiometrics\" : {\"leftEye\" : {\"type\" : \"iris\", \"missingBiometric\" : \"leftEye\",\"reason\" : \"Missing biometrics\",\"exceptionType\" : \"Permanent\",\"individualType\" : \"INDIVIDUAL\"}}}");
+		Map<String, String> tags = exceptionBiometricsTagGenerator.generateTags("12345", "1234", "NEW", null,
+				metaInfoMap, 0);
+		assertEquals("LE", tags.get(tagName));
+	}
+
+	@Test
+	public void testGenerateTagsForExceptionBiometricsNotAvailableWithBCTest()
+			throws BaseCheckedException, JsonMappingException, JsonProcessingException {
+		Map<String, String> metaInfoMap = new HashMap<>();
+		metaInfoMap.put(JsonConstant.EXCEPTIONBIOMETRICS, "{\"applicant\" : {}}");
+		metaInfoMap.put(JsonConstant.METADATA,
+				"[ {\n  \"label\" : \"registrationId\",\n  \"value\" : \"1234\"\n},{\n  \"label\" : \"Registration Client Version Number\",\n  \"value\" : \"1.1.4\"\n} ]");
+		Mockito.when(mapper.readValue(anyString(), any(Class.class)))
+				.thenReturn(new FieldValue("Registration Client Version Number", "1.1.4"));
+		Map<String, String> tags = exceptionBiometricsTagGenerator.generateTags("12345", "1234", "NEW", null,
+				metaInfoMap, 0);
+		assertEquals("", tags.get(tagName));
+	}

```

Comment:
Test cases for consideration
applicant - 1.1.4 - success
applicant - "" - success
applicant - 1.2.0.1 - failure("tag not avilable")
individual biometrics - 1.1.4 - failure("tag not avilable")
individual biometrics  - "" - failure("tag not avilable")
individual biometrics  - 1.2.0.1 - success
"" - 1.1.4 - failure("tag not avilable")
""  - "" - failure("tag not avilable")
"" - 1.2.0.1 - failure("tag not avilable")

Lets refer these combinations and add test cases for cases that is not covered.

Reply:
Checked and added new test cases to cover all above